### PR TITLE
Eng 12454 mismatch uac zk fix

### PR DESCRIPTION
**Pull request by xinjiacs**
Add exclusive access check between node rejoin and UpdateApplicationCatalog using ZK nodes

### CODE REVIEW

**Comment by jhugg** [comment]
This looks ok to me. Maybe go over it with Walter, or someone who knows ZK really well?

File: src/frontend/org/voltdb/rejoin/Iv2RejoinCoordinator.java [hunk 1]
```diff
@@ -181,7 +182,33 @@ private String makeSnapshotRequest(Multimap<Long, Long> sourceToDests)
...
+            VoltZK.createCatalogUpdateBlocker(m_messenger.getZK(), VoltZK.rejoinActiveBlocker);
+
+            if (m_messenger.getZK().exists(VoltZK.uacActiveBlocker, false) == null) {
+                return;
+            }
+
+            // uac zk blocker exists, rejoin node should wait to watch its stat
+            VoltZK.removeCatalogUpdateBlocker(m_messenger.getZK(), VoltZK.rejoinActiveBlocker, REJOINLOG);
+
+            REJOINLOG.info(String.format("Rejoin node is waiting %d seconds for @UpdateApplicationCatalog to finish",
```
Comment (luyangco) [inline src/frontend/org/voltdb/rejoin/Iv2RejoinCoordinator.java]: is waiting --> will wait

File: src/frontend/org/voltdb/sysprocs/UpdateApplicationCatalog.java [hunk 2]
```diff
@@ -519,90 +519,106 @@ private final void performCatalogVerifyWork(
...
+                else {
+                    String errmsg = "Invalid catalog update.  Catalog or deployment change was planned " +
+                            "against one version of the cluster configuration but that version was " +
+                            "no longer live when attempting to apply the change.  This is likely " +
+                            "the result of multiple concurrent attempts to change the cluster " +
+                            "configuration.  Please make such changes synchronously from a single " +
+                            "connection to the cluster.";
+                    log.warn(errmsg);
+
+                    VoltZK.removeCatalogUpdateBlocker(zk, VoltZK.uacActiveBlocker, log);
```
Comment (luyangco) [inline src/frontend/org/voltdb/sysprocs/UpdateApplicationCatalog.java]: Do you need it? You already have the finally block to remove uac blocker.

File: src/frontend/org/voltdb/sysprocs/UpdateApplicationCatalog.java [hunk 3]
```diff
@@ -519,90 +519,106 @@ private final void performCatalogVerifyWork(
...
+                CatalogUtil.updateCatalogToZK(
+                        zk,
+                        catalogStuff.version,
+                        catalogStuff.txnId,
+                        catalogStuff.uniqueId,
+                        catalogStuff.catalogBytes,
+                        catalogStuff.getCatalogHash(),
+                        catalogStuff.deploymentBytes);
+
+                VoltZK.removeCatalogUpdateBlocker(zk, VoltZK.uacActiveBlocker, log);
```
Comment (luyangco) [inline src/frontend/org/voltdb/sysprocs/UpdateApplicationCatalog.java]: Same here.

**Pull request merged**